src/main/java/org/jabref/gui/groups/DeleteDialog.java

```diff
@@ -0,0 +1,125 @@
+package org.jabref.gui.groups;
+
+import javax.swing.JButton;
+import javax.swing.JDialog;
+import javax.swing.JPanel;
+import javax.swing.KeyStroke;
+
+import org.jabref.logic.l10n.Localization;
+import org.jabref.model.groups.GroupTreeNode;
+import javafx.collections.ObservableList;
+
+import javax.swing.BoxLayout;
+import javax.swing.JLabel;
+import javax.swing.JComponent;
+
+import java.awt.event.ActionListener;
+import java.awt.event.KeyEvent;
+import java.util.Iterator;
+import java.util.Set;
+import java.awt.event.ActionEvent;
+
+public class DeleteDialog extends JDialog {
+
+	public enum DeleteResult {
+		DELETE_FROM_DATABASE,
+		REMOVE_FROM_GROUPS,
+		CANCEL
+	}
+
+	protected DeleteResult result = DeleteResult.CANCEL;
+
+	/**
+	 * Create the dialog.
+	 * @param selectedGroups 
+	 * @param allEnriesGroups 
+	 */
+	public DeleteDialog(ObservableList<GroupTreeNode> selectedGroups, Set<String> allEnriesGroups) {
+		setTitle("Delete");
+		setModal(true);
+		getContentPane().setLayout(new BoxLayout(getContentPane(), BoxLayout.Y_AXIS));
+
+		JPanel panel = new JPanel();
+		getContentPane().add(panel);
+
+		String message = "";
```

Comment:
In my opinion this dialog right now is a bit to complicated. My proposal:
Main text: "The entry / entries also belongs to the currently selected group(s). Do you want to remove it from the group or delete it completely?" (maybe with better english :-) )
Options: "Remove from group(s)", "Delete from database", "Cancel"

Reply:
Moreover, I miss a "Remember this choice" checkbox.

Reply:
Hi @tobiasdiez, 
Thanks for comments.
The main purpose of this PR is to notify that some entry could belong to some groups that couldn't be identified by watching on screen (groups that are not selected). If we remove that information the new dialog is useless.

Reply:
Ok, then I don't understand the use case. I thought you wanted to prevent that a user by accident deletes an entry although he just wanted to remove it from the currently selected groups. Why is the information about the other groups important? Moreover, if you select an entry, then groups that contain this entry are highlighted in the group pane so the information is already there (in some form).